### ISSUE TITLE
`switch_case_alignment` failing on alignment in one-line case expressions

### ISSUE THREAD

**Issue by lhunath** [realm/SwiftLint#5373]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

A new syntax is now available for `if` and `switch` allowing for one-line expression statements with these. As such, the following are now completely acceptable and normal:
```
return if a { a } else { b }
return switch a { case .x: a default: b }
```

SwiftLint should be updated to allow for these, eg:
```
warning: Switch and Case Statement Alignment Violation: Case statements should be indented within their closing brace (switch_case_alignment)
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.53.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
included:
  - Hubstaff
excluded:
  - Hubstaff/Modules/Services/Activities/Server/GraphQL # https://github.com/realm/SwiftLint/issues/2329
  - Hubstaff/Modules/Services/Server/GraphQL
  - Hubstaff/App/Tests/Mocks
line_length:
  warning: 140
  ignores_comments: true
  ignores_interpolated_strings: true
colon:
  flexible_right_spacing: true
type_name:
  allowed_symbols: "_"
  min_length: 1
identifier_name:
  allowed_symbols: "_"
  min_length: 1
switch_case_alignment:
  indented_cases: true
function_parameter_count: 8
file_length: 1500
cyclomatic_complexity:
  warning: 15
  ignores_case_statements: true
function_body_length: 80
trailing_comma:
  mandatory_comma: true
nesting:
  type_level: 4
statement_position:
  statement_mode: uncuddled_else
opening_brace:
  allowMultilineFunc: true
disabled_rules:
  - todo              # TODOs are permitted as annotated markers for future improvements.
  - large_tuple       # We allow large tuples as cheap and easy anonymous struct substitutes.
  - opening_brace     # https://github.com/realm/SwiftLint/issues/4026
  - type_body_length  # https://github.com/realm/SwiftLint/issues/4631
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? no
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 15.0.1 Build version 15A507

```swift
return switch a { case .x: a default: b }
```

**Comment by SimplyDanny** [comment]
Do you suggest a new option to enable this behavior?

**Comment by lhunath** [comment]
personally I feel that `return switch a { case .x: a default: b }` shouldn't trigger a lint by default as a lint warning seems in conflict with the intent behind the new Swift expression syntax, which is to allow for expressive conditionals in minimal code-space.

**Comment by SimplyDanny** [comment]
I don't think the main intend is to get minimal code-space, but to allow code without much noise. So I think that

```swift
func f() -> Int {
    switch thing {
    case .x: 1
    default: 2
    }
}
```

is also in the sense of this new language feature and I guess there are many folks who like this style more than a one-liner.

That said, we could go like: If the expression occupies multiple lines, the normal behavior of the rule is applied. A new option could disable the rule for expressions that only span a single line.

**Comment by tahirmt** [comment]
There's also a failure when switch is used as an expression for assignment

```swift
let someValue: String = switch something {
case .first: "first"
case .second: "second"
}
```

**Comment by SimplyDanny** [comment]
> There's also a failure when switch is used as an expression for assignment
> 
> ```swift
> let someValue: String = switch something {
> case .first: "first"
> case .second: "second"
> }
> ```

That should not be the case with the latest release of SwiftLint. Are you using 0.54.0, @tahirmt?

**Comment by tahirmt** [comment]
Oh I was on 0.51.0. Thanks
